sdk/keyvault/azure-security-keyvault-keys/inc/azure/keyvault/keys/key_client.hpp

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <azure/core/credentials.hpp>
+#include <azure/core/http/http.hpp>
+#include <azure/core/response.hpp>
+#include <azure/keyvault/common/keyvault_pipeline.hpp>
+
+#include "azure/keyvault/keys/key_client_options.hpp"
+#include "azure/keyvault/keys/key_vault_key.hpp"
+
+#include <functional>
+#include <vector>
+
+namespace Azure { namespace Security { namespace KeyVault { namespace Keys {
+
+  namespace Details {
+    constexpr static const char* KeysPath = "keys";
+    constexpr static const char* DeletedKeysPath = "deletedkeys";
+  } // namespace Details
+
+  /**
+   * @brief The KeyClient provides synchronous methods to manage a KeyVaultKe in the Azure Key
+   * Vault. The client supports creating, retrieving, updating, deleting, purging, backing up,
+   * restoring, and listing the KeyVaultKey.
+   */
+  class KeyClient {
+  protected:
+    std::unique_ptr<Azure::Security::KeyVault::Common::Internal::KeyVaultPipeline> m_pipeline;
```

Comment:
This should just be a Core pipeline

Reply:
He addressed it above, but FWIW, in .NET we have our own wrapper for reasons he stated, but also to better handle a few non-exceptional cases that return errors from the service but we don't want to throw, which the Core pipeline will. If used similarly, I think it's worth it to have your own wrapper here.

Reply:
Talked offline about this to Jeff. He agreed to use the KeyVault pipeline strategy